rundeckapp/grails-app/conf/spring/resources.groovy

```diff
@@ -527,7 +529,22 @@ beans={
         defaultConverters=['StorageTimestamperConverter','KeyStorageLayer']
         loggerName='org.rundeck.storage.events'
     }
```

Comment:
I think you can remove the original `rundeckStorageTreeFactory` bean definition, and pass the startup "config" data to the creator, maybe as another Map value.  the "createOnStartup" could use the normal create() but pass in the startup config